vegafusion-core/src/task_graph/memory.rs

```diff
@@ -94,3 +95,12 @@ pub fn inner_size_of_table(value: &VegaFusionTable) -> usize {
     let size_of_batches: usize = value.batches.iter().map(size_of_record_batch).sum();
     schema_size + size_of_batches
 }
+
+pub fn inner_size_of_logical_plan(plan: &LogicalPlan) -> usize {
+    size_of_val(&plan)
```

Comment:
since plan is of type `&LogicalPlan` using `&plan` will give size of the pointer (i.e. `&&plan`) and not the size of the LogicalPlan struct itself
```suggestion
    size_of_val(plan)
```